### PR TITLE
mgmt, local generation for Compute 2023-10-02

### PR DESCRIPTION
**Pull request by v-hongli1**
relate to [issue#4850](https://github.com/Azure/sdk-release-request/issues/4850)

# Description

Fixed `PlatformImageNotFound` exception for test case `VirtualMachinePopularImageTests#canCreateAllPopularImageVM`.

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachinePopularImageTests.java [hunk 1]
```diff
@@ -31,7 +31,9 @@ public void canCreateAllPopularImageVM() {
         rgName = generateRandomResourceName("rg", 10);
         List<Mono<VirtualMachine>> vmMonos = new ArrayList<>();
         for (KnownWindowsVirtualMachineImage image : Arrays.stream(KnownWindowsVirtualMachineImage.values())
-            .filter(image -> image != KnownWindowsVirtualMachineImage.WINDOWS_SERVER_2019_DATACENTER_WITH_CONTAINERS_GEN2 && image != KnownWindowsVirtualMachineImage.WINDOWS_SERVER_2019_DATACENTER_WITH_CONTAINERS)
+            .filter(image -> image != KnownWindowsVirtualMachineImage.WINDOWS_SERVER_2019_DATACENTER_WITH_CONTAINERS_GEN2
+                && image != KnownWindowsVirtualMachineImage.WINDOWS_SERVER_2019_DATACENTER_WITH_CONTAINERS
+                && image != KnownWindowsVirtualMachineImage.WINDOWS_DESKTOP_10_20H1_PRO)
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachinePopularImageTests.java:36]: Does previous api-version fail this test too?

Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachinePopularImageTests.java:36]: If these images no longer supported by Compute, you need to give proper javadoc in code. e.g. https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/KnownLinuxVirtualMachineImage.java#L22-L27
and see if there is a replacement.

Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachinePopularImageTests.java:36]: Yes, the test case `VirtualMachinePopularImageTests#canCreateAllPopularImageVM` fails in `RECORD` and `LIVE` modes.

Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachinePopularImageTests.java:36]: The comment `@Deprecated` has been fixed in the new version.

File: sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/KnownLinuxVirtualMachineImage.java [hunk 2]
```diff
@@ -21,20 +21,26 @@ public enum KnownLinuxVirtualMachineImage {
 
     /**
      * OpenSUSE-Leap 15.1.
-     * @deprecated use OPENSUSE_LEAP_15.
+     * @deprecated use OPENSUSE_LEAP_15_1.
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/KnownLinuxVirtualMachineImage.java:24]: 15_1 is also deprecated?

File: sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/KnownLinuxVirtualMachineImage.java [hunk 3]
```diff
@@ -21,20 +21,26 @@ public enum KnownLinuxVirtualMachineImage {
...
      */
     @Deprecated
     OPENSUSE_LEAP_15_1("SUSE", "openSUSE-Leap-15-3", "gen1"),
-    /** OpenSUSE-Leap 15. */
+    /** OpenSUSE-Leap 15.
+     * @deprecated use OPENSUSE_LEAP_15.
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/KnownLinuxVirtualMachineImage.java:29]: I don't quite understand the deprecation message here.

File: sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/KnownLinuxVirtualMachineImage.java [hunk 4]
```diff
@@ -21,20 +21,26 @@ public enum KnownLinuxVirtualMachineImage {
...
+     */
+    @Deprecated
     OPENSUSE_LEAP_15("SUSE", "openSUSE-Leap-15-3", "gen1"),
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/KnownLinuxVirtualMachineImage.java:32]: Please check if there's a minor version update(e.g. `openSUSE-Leap-15-4`). We don't want to deprecate major version directly.

File: sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/KnownLinuxVirtualMachineImage.java [hunk 5]
```diff
@@ -21,20 +21,26 @@ public enum KnownLinuxVirtualMachineImage {
...
+     * @deprecated use SLES_15_SP1.
      */
     @Deprecated
     SLES_15_SP1("SUSE", "sles-15-sp2", "gen1"),
-    /** SLES 15. */
+    /** SLES 15.
+     * * @deprecated use SLES_15.
+     */
+    @Deprecated
     SLES_15("SUSE", "sles-15-sp2", "gen1"),
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/KnownLinuxVirtualMachineImage.java:44]: Same here.

Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/KnownLinuxVirtualMachineImage.java]: Fixed in the new version.

**Comment by v-hongli1** [comment]
Added `<revapi.skip>true</revapi.skip>` in `pom.xml` for fix ci analyze exception. 
The detail infomation following as:
https://dev.azure.com/azure-sdk/public/_build/results?buildId=3390320&view=results

File: sdk/resourcemanager/azure-resourcemanager-compute/CHANGELOG.md [hunk 6]
```diff
@@ -1,14 +1,12 @@
 # Release History
 
-## 2.35.0-beta.1 (Unreleased)
+## 2.35.0-beta.1 (2024-01-03)
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-compute/CHANGELOG.md:3]: Don't change this date

**Review by weidongxu-microsoft — COMMENTED** [review]
One comment.

LGTM

File: sdk/resourcemanager/azure-resourcemanager-compute/CHANGELOG.md (hunk 6 above)
Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-compute/CHANGELOG.md]: Fixed in the new version.

**Review by weidongxu-microsoft — APPROVED** [review]
LGTM. Please wait for Xiaofei's approval.

In the end, handwritten code change is on the 2 Known###VirtualMachineImage, correct?

**Comment by v-hongli1** [comment]
Yes, the handwritten code change is only on the 2 Known###VirtualMachineImage.

File: sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/KnownWindowsVirtualMachineImage.java [hunk 7]
```diff
@@ -5,7 +5,7 @@
 /** The popular Azure Windows images. */
 public enum KnownWindowsVirtualMachineImage {
     /** Windows 10 2020 H1 Pro */
-    WINDOWS_DESKTOP_10_20H1_PRO("MicrosoftWindowsDesktop", "Windows-10", "20h1-pro"),
+    WINDOWS_DESKTOP_10_20H1_PRO("MicrosoftWindowsDesktop", "Windows-10", "win10-22h2-pro"),
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/KnownWindowsVirtualMachineImage.java:8]: @weidongxu-microsoft Should we deprecate this enum and provide an alternative major version `WINDOWS_DESKTOP_10_PRO `?

Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/KnownWindowsVirtualMachineImage.java:8]: Ah, yes, we should do this.

The name of the enum is no longer consistent with its content.

File: sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/KnownWindowsVirtualMachineImage.java [hunk 8]
```diff
@@ -4,7 +4,12 @@
 
 /** The popular Azure Windows images. */
 public enum KnownWindowsVirtualMachineImage {
-    /** Windows 10 2020 H1 Pro */
+    WINDOWS_DESKTOP_10_PRO("MicrosoftWindowsDesktop", "Windows-10", "win10-22h2-pro"),
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/KnownWindowsVirtualMachineImage.java:7]: Probably need to move to the end of enum. Order change of enum constants would be considered breaking change for CI.

Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/KnownWindowsVirtualMachineImage.java:7]: Also javadocs.

Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/KnownWindowsVirtualMachineImage.java]: Fixed in the new version.

**Review by XiaofeiCao — APPROVED** [review]
May need rerun failed CI.

**Pull request merged**